src/capabilities.ts

```diff
@@ -21,3 +21,11 @@ export interface Capabilities {
      */
     capabilities(request: CapabilitiesRequest): Promise<GenericResponse>;
 }
+
+export interface PartialCapabilityStatement {
+    [resourceType: string]: any;
+}
+
+export interface CapabilityRegistry {
+    getCapabilities(): PartialCapabilityStatement;
+}

```

Comment:
I think that `CapabilityRegistry` shouldn't be part of the interface package. It's best to minimize the number of interfaces, and only introduce new ones when they are required for collaboration between packages/modules.

It looks like this interfaced is only used inside of the routing package. Similar to how [FHIRSearchParametersRegistry](https://github.com/awslabs/fhir-works-on-aws-search-es/blob/feat-search-igs/src/FHIRSearchParametersRegistry/index.ts) is internal to the search package.

It's a bit weird to have to build a `CapabilityRegistry` on the deployment package to then pass it as an argument to the router when the router is the only module that uses such registry.

**My recommendation is:**
Add a field to the [FhirConfig.Profile](https://github.com/awslabs/fhir-works-on-aws-interface/blob/08ada005426132377a65860e0faa9870fa6c0ad7/src/fhirConfig.ts#L106) like `compiledImplementationGuides?: any`. And have the router internally build a `CapabilityRegistry` and use it as appropiate. This is similar to how the [Search package takes compiledImplementationGuides as a constructor argument](https://github.com/awslabs/fhir-works-on-aws-search-es/blob/623c75af291c7c795b263a785f1d73a1dcffd066/src/elasticSearchService.ts#L70).

This minimizes the changes to the interface and makes it backwards compatible, since the new `FhirConfig.Profile.compiledImplementationGuides` field is optional.

Reply:
I looked at the other PRs related to this change and overall the implementation looks good. Making the recommended interface change shouldn't have a huge impact on the other PRs. It's mostly the same code, just moving a few more parts of it into the routing package.

Reply:
@carvantes thanks for the review and the feedback. How would you suggest we handle the following scenario: 

Since the `FHIRStructureDefinitionRegistry` needs to be passed the path to `compiledImplementationGuides/` as an argument, only the deployment package would be in a position to know that path. Hence the original need build this instance of a`CapabilityRegistry` in the deployment package. 

Separately, is there any interest in having all the registries conform to the `CapabilityRegistry` interface, including `FHIRSearchParametersRegistry`?

Reply:
Looking at the other PRs, `FHIRStructureDefinitionRegistry` takes the `compiledImplementationGuides` object as an argument. It is not aware of the `compiledImplementationGuides/` folder at all.
https://github.com/awslabs/fhir-works-on-aws-routing/blob/ecf7249f7d0310b8d05d4fe773afd901937fc393/src/registry/index.ts#L7
That's why I thought it'd be easy to move up the `compiledImplementationGuides` parameter into `FhirConfig.Profile.compiledImplementationGuides`. Am I missing something?

`FHIRSearchParametersRegistry` is internal to the search package, so it does not really matter what interfaces it implements. The `Search` interface is the one that already declares a `getCapabilities` method.
https://github.com/awslabs/fhir-works-on-aws-interface/blob/mainline/src/search.ts#L75

I don't think that we should standardize on a `CapabilityRegistry` interface. I think it's best to follow the pattern where the main fwoa interfaces (Search, Peristence, Auth) somehow expose their capabilities and then the Router combines everything into a Capability statement. It's ok to have different interfaces to expose capabilities, i.e. `Search` has `getCapabilities()` because it needs to specify search params for each resource,  but `Persistence` simply has a boolean field `updateCreateSupported` because that applies to all resources.
https://github.com/awslabs/fhir-works-on-aws-interface/blob/mainline/src/persistence.ts#L49

Reply:
@carvantes Sorry, you're right; I got the `FHIRStructureDefinitionRegistry` constructor args mixed up. This makes more sense to me now. I will make the changes today.

Reply:
@carvantes I updated this MR as discussed.